### ISSUE TITLE
Diving deeper

### ISSUE THREAD

**Issue by dirkroorda** [glottobank/cldf#24]
May I point you to a more profound way to deal with tabular data?
Core idea: column storage with support for sparse data.
Step one: define anchor points for the most granular units in your corpus.
Step two: define nodes for all subsets of anchor points that you want to annotate.
Step three: annotate in the form of mappings from nodes to feature values.
Every feature is 1 column of data, implicitly linked to the set of anchor points and nodes.
I have used this model extensively for the Hebrew Bible (400,000 anchor points, 1,000,000 extra nodes, 20,000,000 feature values), and it enables very flexible explorative data processing, e.g. in a Jupyter notebook.
By isolating columns, it becomes easier to produce them in a distributed fashion, and share them easily.
Have a look at the [model](https://github.com/ETCBC/text-fabric/wiki/Data-model)

**Comment by xrotwang** [comment]
Thanks for the pointer! I guess the subset of "dealing with tabular data" we want to address with CLDF is rather narrow - basically serializing it in a portable way. It seems as if text-fabric also comprises this aspect via its [tf format](https://github.com/ETCBC/text-fabric/wiki/File-formats)? But I'm not sure you actually propose this format as general data exchange format for linguistic data.

Anyway, I'd think as long as we don't come up with formats which are inherently incompatible with analysis frameworks like text-fabric, our efforts are mostly orthogonal.

**Comment by dirkroorda** [comment]
Right. In a sense, converting into TF is not that difficult. A fellow and I converted New Testament Treebanks into TF, and all it took was day of joint programming. We retained all information of the source.
And last Saturday I converted a Sanskrit corpus to TF, during a train trip from Leipzig back to the Netherlands. 

Recent experience pointed out to me the value of separating columns. Quite often, features correspond to fields of experience. Semantics, syntax, pragmatics, AI, etc. This experience is located in different research groups. So if they can package their expertise in feature modules, publish them on github, then others can retrieve them, juxtapose them, and work with interesting combinations of features. So a column-organization of data, such as TF has, really supports sharing of data.
